Add CalculatedSummary block type

dwyeradam opened the pull request:
### What is the context of this PR?
As a user
I want my figures I have entered to be presented to me as a total
So that I don't have to provide a total as well as a breakdown and I have the opportunity to check for any mistakes

### How to review 
test_calculated_summary.json

### Checklist

* [x] New static content marked up for translation
* [x] Newly defined schema content included in eq-translations repo

dwyeradam committed:
Add CalculatedSummary block type

benhowes commented on app/templating/view_context.py:
```diff
@@ -88,39 +97,148 @@ def build_view_context_for_question(metadata, schema, answer_store, current_loca
+def _get_formatted_total(block, answer_store, group_instance, schema):
+    calculated_total = 0
+    for answer in block['calculation']['answers_to_calculate']:
+        if group_instance > 0 and not schema.answer_is_in_repeating_group(answer):
```
Regarding behaviour with repeats:
1) Do we know that this should always be for the current `group_instance`? 
2) This will not return results a value for repeating answers either?

I can't see anything in the ticket/acceptance criteria around this, so thought we should clear up the expected behaviour

benhowes replied:
As discussed, there's no user need for this, so this is fine until the requirements change, if ever

benhowes reviewed (commented):
Looks good, tested locally with no problems, really obvious test schema! Just one point to clarify which may well not require any test changes.

jonnyshaw89 commented on app/templating/view_context.py:
```diff
@@ -88,39 +97,148 @@ def build_view_context_for_question(metadata, schema, answer_store, current_loca
+            blocks.append(_remove_unwanted_questions_answers(block, answers_in_block, answer_matches))
+
+    section = {
+        'title': '',
```
Why is this empty?

jonnyshaw89 commented on app/templating/view_context.py:
```diff
@@ -88,39 +97,148 @@ def build_view_context_for_question(metadata, schema, answer_store, current_loca
+        'groups': [
+            {
+                'id': group_id,
+                'title': '',
```
and this?
should either have content or not be present

dwyeradam replied:
group.py requires 'title' key but I don't want to use it on the page. Any text here would show under the Boxed "Please review your answer and confirm these are correct."
I can get rid of the empty section title though.

dwyeradam commented:
I hope you don't hold this against me and considering it was my retro point but I squashed as I rebased. It was only deleting a single line after all.

dcdarrell9 commented on app/templating/view_context.py:
```diff
@@ -88,39 +97,147 @@ def build_view_context_for_question(metadata, schema, answer_store, current_loca
+
+        answer_matches = set(answers_in_block.keys()) & set(answers_to_calculate)
+        if answer_matches:
+            blocks.append(_remove_unwanted_questions_answers(block, answers_in_block, answer_matches))
```
is it worth maybe doing something like `blocks = [_remove_unwanted_questions_answers(block, answers_in_block, answer_matches)]`

dwyeradam replied:
This is within a loop. If I did that blocks would only contain the last block in the loop.

dcdarrell9 reviewed (approved):
LGTM
